src/Umbraco.Web.UI.Client/src/views/propertyeditors/grid/dialogs/config.controller.js

```diff
@@ -17,21 +17,20 @@ function ConfigController($scope) {
         }
     }
 
-    vm.showEmptyState = function() {
-        return $scope.model.config.length === 0
-            && Object.keys($scope.model.config).length === 0
-            && $scope.model.styles.length === 0;
-    }
+    vm.showEmptyState = false;
+    vm.showConfig = false;
+    vm.showStyles = false;
 
-    vm.showConfig = function() {
-        return $scope.model.config &&
-            ($scope.model.config.length > 0 || Object.keys($scope.model.config).length > 0);
-    }
+    $scope.$watchGroup(['model.config.length', 'model.styles.length'], function(newValues, oldValues, scope) {
```

Comment:
Won't this revert reacting to the cases where model.config is a hash? That's the whole point of the PR. 🤪

Reply:
I just tested it with both configs and it seems to work. 🤯
In the "old" days, wasn't an object's length equal to it's count of keys?
If so, I don't really see why @bjarnef's changes ended up hiding everything, but oh well. :P

Reply:
@lars-erik I think you are right, this is not the right way of covering that case. So I have switched it to a watchCollection() instead. This can handle both Array or Object, and will watch for shallow changes, aka. the first level (Keys or Items)

Reply:
@lars-erik could I ask you to test this against your package — that would simplify the task of the reviewer for this. Thanks :-)

Reply:
I did test it with this, and I realized I mutate the settings before they arrive in this controller. After that I only mutate a nested setting thing, but that's not affected by this. However, switching to watchCollection would allow for even more dynamic settings which could be good. I do show/hide settings via a custom property editor. I don't think a deep watch of a settings object/collection will cost that much performance.